src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -79,6 +80,8 @@
 
   private static final int COMMENT_LINE_LENGTH = 75;
 
+  private static final Pattern versionPattern = Pattern.compile("v\\d+(p\\d)?beta");
```

Comment:
Would it make sense to just do something like strings.Contains(str, "beta")? Since the \d+ is optional, it's probably OK to just remove.

WDYT?

Reply:
Since google APIs follow the versioning scheme defined in https://goto.google.com/aip/0201, I figured we could be extra certain that a package name contains a version string representing beta if we matched against those guidelines.

Reply:
Gotcha, thanks for the context. I'll leave that call to y'all!

Reply:
I'd just do `v\\d.*beta`. I think the current pattern will have trouble with `v1p11beta1`.

Reply:
haha I never get these regex patterns right the first time
Done.